File: v2/HarvestAPISample.java
import java.io.*;
import java.net.*;

public class HarvestAPISample {
  public static void main(String[] args) throws IOException, ProtocolException {
    URL url = new URL("https://api.harvestapp.com/v2/users/me");
    HttpURLConnection conn = (HttpURLConnection) url.openConnection();

    conn.setRequestMethod("GET");
    conn.setRequestProperty("Accept", "application/json");
    conn.setRequestProperty("User-Agent", "Java Harvest API Sample");
    conn.setRequestProperty("Authorization", "Bearer " + System.getenv("HARVEST_ACCESS_TOKEN"));
    conn.setRequestProperty("Harvest-Account-ID", System.getenv("HARVEST_ACCOUNT_ID"));

    String inputLine;
    BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
    StringBuffer response = new StringBuffer();

    while ((inputLine = in.readLine()) != null) {
      response.append(inputLine);
    }

    if (conn.getResponseCode() == 200) {
      System.out.println(response);
    } else {
      System.out.println("Error " + conn.getResponseCode() + ": " + response);
    }

    in.close();
    conn.disconnect();
  }
}

File: v2/harvest_api_sample.go
package main

import "fmt"
import "io/ioutil"
import "encoding/json"
import "net/http"
import "os"

func main() {
  url := "https://api.harvestapp.com/v2/users/me"
  client := &http.Client{}

  req, _ := http.NewRequest("GET", url, nil)
  req.Header.Set("User-Agent", "Go Harvest API Sample")
  req.Header.Set("Harvest-Account-ID", os.Getenv("HARVEST_ACCOUNT_ID"))
  req.Header.Set("Authorization", "Bearer " + os.Getenv("HARVEST_ACCESS_TOKEN"))

  resp, _ := client.Do(req)
  body, _ := ioutil.ReadAll(resp.Body)
  defer resp.Body.Close()

  var jsonResponse map[string]interface{}

  json.Unmarshal(body, &jsonResponse)

  prettyJson, _ := json.MarshalIndent(jsonResponse, "", "  ")
	fmt.Println(string(prettyJson))
}
